### PR TITLE
Use ExpectEqual in test/e2e/framework

### LINKED ISSUE #79686: Use framework.ExpectEqual() for e2e tests
**What happened**:

After fixing golint failures under `test/e2e`, we need to put `gomega` for each `Expect()` and `Equal()`.
Then common error checking code has become like:

`gomega.Expect(foo).To(gomega.Equal(bar))`

We can replace this code with `framework.ExpectEqual(foo, bar)` which is more readable and easy to be understood what the code does.

Now we have following e2e test directories which can be replaced:
- [x] test/e2e/apimachinery (https://github.com/kubernetes/kubernetes/pull/79685)
- [x] test/e2e/apps (https://github.com/kubernetes/kubernetes/pull/79695)
- [x] test/e2e/auth (https://github.com/kubernetes/kubernetes/pull/79762)
- [x] test/e2e/autoscaling (https://github.com/kubernetes/kubernetes/pull/79765)
- [x] test/e2e/cloud (https://github.com/kubernetes/kubernetes/pull/79821)
- [x] test/e2e/common (https://github.com/kubernetes/kubernetes/pull/79769)
- [ ] test/e2e/framework (https://github.com/kubernetes/kubernetes/pull/79771)
- [x] test/e2e/kubectl (https://github.com/kubernetes/kubernetes/pull/79699)
- [x] test/e2e/lifecycle (https://github.com/kubernetes/kubernetes/pull/79860)
- [x] test/e2e/network (https://github.com/kubernetes/kubernetes/pull/79791)
- [x] test/e2e/node (https://github.com/kubernetes/kubernetes/pull/79792)
- [x] test/e2e/scalability (https://github.com/kubernetes/kubernetes/pull/79842)
- [x] test/e2e/scheduling (https://github.com/kubernetes/kubernetes/pull/79781)
- [x] test/e2e/servicecatalog (https://github.com/kubernetes/kubernetes/pull/79754)
- [x] test/e2e/storage (https://github.com/kubernetes/kubernetes/pull/79755)
- [x] test/e2e/upgrades (https://github.com/kubernetes/kubernetes/pull/79756)
- [x] test/e2e/windows (https://github.com/kubernetes/kubernetes/pull/79756)
- [x] test/e2e_node/apparmor_test.go (https://github.com/kubernetes/kubernetes/pull/80790)
- [x] test/e2e_node/container_log_rotation_test.go (https://github.com/kubernetes/kubernetes/pull/80790)
- [x] test/e2e_node/container_manager_test.go (https://github.com/kubernetes/kubernetes/pull/80790)
- [x] test/e2e_node/critical_pod_test.go (https://github.com/kubernetes/kubernetes/pull/80790)
- [x] test/e2e_node/device_plugin_test.go (https://github.com/kubernetes/kubernetes/pull/80790)
- [x] test/e2e_node/eviction_test.go (https://github.com/kubernetes/kubernetes/pull/80843)
- [x] test/e2e_node/gpu_device_plugin_test.go (https://github.com/kubernetes/kubernetes/pull/80843)
- [x] test/e2e_node/mirror_pod_test.go (https://github.com/kubernetes/kubernetes/pull/80843)
- [x] test/e2e_node/util.go (https://github.com/kubernetes/kubernetes/pull/80843)
- [x] test/e2e_kubeadm/dns_addon_test.go (https://github.com/kubernetes/kubernetes/pull/80795)
- [x] test/e2e_kubeadm/kubeadm_certs_test.go (https://github.com/kubernetes/kubernetes/pull/80795)
- [x] test/e2e_kubeadm/proxy_addon_test.go (https://github.com/kubernetes/kubernetes/pull/80795)

### PR DESCRIPTION
**Pull request by ZP-AlwaysWin**
What type of PR is this?
/kind cleanup
/priority backlog
/assign @oomichi

What this PR does / why we need it:
use framework.ExpectEqual in test/e2e/framework

Which issue(s) this PR fixes:

Does this PR introduce a user-facing change?:

NONE

### CODE REVIEW

**Review by draveness — APPROVED** [review]
/ok-to-test
/release-note-none
/lgtm

File: test/e2e/framework/pod/resource.go [hunk 1]
```diff
@@ -672,13 +673,13 @@ func GetPodsScheduled(masterNodes sets.String, pods *v1.PodList) (scheduledPods,
 		if !masterNodes.Has(pod.Spec.NodeName) {
 			if pod.Spec.NodeName != "" {
 				_, scheduledCondition := podutil.GetPodCondition(&pod.Status, v1.PodScheduled)
-				gomega.Expect(scheduledCondition != nil).To(gomega.Equal(true))
-				gomega.Expect(scheduledCondition.Status).To(gomega.Equal(v1.ConditionTrue))
+				framework.ExpectEqual(scheduledCondition != nil,true)
```
Comment (andrewsykim) [inline test/e2e/framework/pod/resource.go]: `ExpectNotEqual(scheduledCondition, nil)`?

File: test/e2e/framework/pod/resource.go [hunk 2]
```diff
@@ -672,13 +673,13 @@ func GetPodsScheduled(masterNodes sets.String, pods *v1.PodList) (scheduledPods,
...
+				framework.ExpectEqual(scheduledCondition.Status, v1.ConditionTrue)
 				scheduledPods = append(scheduledPods, pod)
 			} else {
 				_, scheduledCondition := podutil.GetPodCondition(&pod.Status, v1.PodScheduled)
-				gomega.Expect(scheduledCondition != nil).To(gomega.Equal(true))
-				gomega.Expect(scheduledCondition.Status).To(gomega.Equal(v1.ConditionFalse))
+				framework.ExpectEqual(scheduledCondition != nil,true)
```
Comment (andrewsykim) [inline test/e2e/framework/pod/resource.go]: ExpectNotEqual(scheduledCondition, nil)?

File: test/e2e/framework/timer/timer_test.go [hunk 3]
```diff
@@ -17,6 +17,7 @@ limitations under the License.
 package timer
 
 import (
+	"k8s.io/kubernetes/test/e2e/framework"
```
Comment (andrewsykim) [inline test/e2e/framework/timer/timer_test.go]: Split this import into separate group below

**Comment by andrewsykim** [comment]
Squash commits please :)

**Comment by ZP-AlwaysWin** [comment]
> Squash commits please :)
hi @andrewsykim 

Thanks your reviews,but I've changed a lot of code, . k8s-ci-robot check it error。Do you know what's going on? 

I tried many times, but I couldn't solve the problem.

File: test/e2e/framework/timer/timer_test.go (hunk 3 above)
Comment (ZP-AlwaysWin) [reply test/e2e/framework/timer/timer_test.go]: What does this mean? I can't pass the k8s-ci-robot check. Can you help me? and I don't know where is error。

**Comment by ZP-AlwaysWin** [comment]
hi @oomichi

I want to change the issues，  but I can't pass the k8s-ci-robot check. Can you help me? and I don't know where is error。

File: test/e2e/framework/pod/resource.go [hunk 4]
```diff
@@ -36,6 +36,7 @@ import (
 	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
 	"k8s.io/kubernetes/pkg/client/conditions"
 	kubepod "k8s.io/kubernetes/pkg/kubelet/pod"
+	e2efwk "k8s.io/kubernetes/test/e2e/framework"
```
Comment (fejta) [inline test/e2e/framework/pod/resource.go]: This is pretty much an abomination as far as I'm concerned. If every import needs its own name... then you're doing a TERRIBLE job naming your packages. I'm going to stop approving PRs that do this.

/assign @timothysc

Comment (andrewsykim) [reply test/e2e/framework/pod/resource.go]: fwiw I don't think `framework` needs an import alias, we only end up import aliasing packages like `test/e2e/framework/node` that is likely to conflict with other packages.

Comment (timothysc) [reply test/e2e/framework/pod/resource.go]: the only reason to import alias framework libraries is the collision with existing names.  

Framework itself should not need this.

**Review by timothysc — CHANGES_REQUESTED** [review]

**Comment by oomichi** [comment]
@ZP-AlwaysWin https://github.com/kubernetes/kubernetes/pull/79771 is the same PR and it has been posted before this.

**Pull request closed without merging**